k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -123,6 +141,23 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
               }
             })
             val watchStream = Json.readStream[W](rsp.reader, Api.BufSize)
+              // it would be nice to not have to flat map over the watch stream, but this is
+              // necessary to handle the K8s bug where erorrs have the incorrect status code.
+              // hopefully this hack can eventually be removed...
+              .flatMap {
+                // special case to handle Kubernetes bug where "too old resource version" errors
+                // are returned with status code 200 rather than status code 410.
+                // see https://github.com/kubernetes/kubernetes/issues/35068 for details.
+                case e: Watch.Error[O] if e.status.message.exists(_.startsWith("too old resource version")) =>
```

Comment:
Can we check `e.status.code` for `410` instead of inspecting the message?

Reply:
That's definitely better than string comparison from a performance perspective. I felt like inspecting the message might be more future-proof – what if Kubernetes were to change the status code returned for this event, or added additional events with status code 410 that should be handled differently – but if you don't think that's as big a concern, I can change this?

Reply:
I think that looking for a code is actually more future proof than looking at a human readable string.  Plus it's already the case that we look for the HTTP status code 410.